  @Test
  public void test() throws Exception {
    Logger logger = mock(Logger.class);

    ProgressReport report = new ProgressReport(ProgressReport.class.getName(), 500, logger);
    report.message("progress");
    report.start("foo start");
    Thread.sleep(800);
    report.stop("foo stop");
    Thread.sleep(300);

    ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
    verify(logger, atLeast(3)).info(captor.capture());

    List<String> messages = captor.getAllValues();
    assertThat(messages.size()).isGreaterThanOrEqualTo(3);
    assertThat(messages.get(0)).isEqualTo("foo start");
    for (int i = 1; i < messages.size() - 1; i++) {
      assertThat(messages.get(i)).isEqualTo("progress");
    }
    assertThat(messages.get(messages.size() - 1)).isEqualTo("foo stop");
  }

